### PR TITLE
Player Docs: Spin Attack

### PR DESCRIPTION
**Pull request by engineer124**
Mainly focuses on the control stick buffers and names `PLAYER_STATE1_CHARGING_SPIN_ATTACK`

### CODE REVIEW

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 1]
```diff
@@ -5564,10 +5567,10 @@ void func_808335B0(PlayState* play, Player* this) {
...
 
 s8 D_8085D090[] = {
-    PLAYER_MWA_STAB_1H,
-    PLAYER_MWA_RIGHT_SLASH_1H,
-    PLAYER_MWA_RIGHT_SLASH_1H,
-    PLAYER_MWA_LEFT_SLASH_1H,
+    PLAYER_MWA_STAB_1H,        // PLAYER_STICK_DIR_FORWARD
+    PLAYER_MWA_RIGHT_SLASH_1H, // PLAYER_STICK_DIR_LEFT
+    PLAYER_MWA_RIGHT_SLASH_1H, // PLAYER_STICK_DIR_BACKWARD
+    PLAYER_MWA_LEFT_SLASH_1H,  // PLAYER_STICK_DIR_RIGHT
```
Comment (engineer124) [inline src/overlays/actors/ovl_player_actor/z_player.c:5573]: Worth noting that the left/right directions don't line up here, so maybe it's worth taking another look at MWA soon

File: src/code/z_camera.c [hunk 2]
```diff
@@ -452,10 +452,10 @@ s32 Camera_IsChargingSwordOrDekuFlowerDive(Camera* camera) {
...
     if (focalActor == &GET_PLAYER(camera->play)->actor) {
         // Charging Sword
-        ret = !!(((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_1000);
+        ret = (((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_CHARGING_SPIN_ATTACK) != 0;
         if (!ret) {
             // Deku Flower Dive
-            ret = !!(((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_100);
+            ret = (((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_100) != 0;
         }
         return ret;
```
Comment (hensldm) [inline src/code/z_camera.c:460]:
```suggestion
                return (((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_CHARGING_SPIN_ATTACK) || (((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_100);
```

You can remove the temp and `!= 0` by just returning the boolean statement.

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 3]
```diff
@@ -8119,34 +8123,36 @@ s32 func_80839770(Player* this, PlayState* play) {
...
         return true;
     }
     return false;
 }
 
-void func_80839860(Player* this, PlayState* play, s32 arg2) {
+void func_80839860(Player* this, PlayState* play, s32 controlStickDirection) {
     s32 pad;
-    f32 speed = (!(arg2 & 1) ? 5.8f : 3.5f);
+    f32 speed = (!(controlStickDirection & 1) ? 5.8f : 3.5f);
```
Comment (hensldm) [inline src/overlays/actors/ovl_player_actor/z_player.c:8136]: Might be worth a comment that `controlStickDirection & 1` is checking if stick direction is either `left/right` or `forward/backward`

**Review by hensldm — APPROVED** [review]

File: src/overlays/actors/ovl_player_actor/z_player.c (hunk 1 above)
Comment (hensldm) [reply src/overlays/actors/ovl_player_actor/z_player.c:5573]: I could be mistaken, but
```c
Player_PlaySfx(this, ((controlStickDirection << 0xE) == (PLAYER_STICK_DIR_BACKWARD << 0xE)) ? NA_SE_PL_ROLL : NA_SE_PL_SKIP);
```
also seems a little strange right?

Comment (AngheloAlf) [reply src/overlays/actors/ovl_player_actor/z_player.c:5573]: Why not add a few TODO comments noting this inconsistency?

Comment (engineer124) [reply src/overlays/actors/ovl_player_actor/z_player.c:5573]: I'm not sure it's that inconsistent. This function deals with either a side hop (which plays `NA_SE_PL_SKIP`) or a backflip (which reuses the sfx `NA_SE_PL_ROLL`). `NA_SE_PL_ROLL` is also used for a regular roll forward, so it's just a multi-use sfx

Comment (hensldm) [reply src/overlays/actors/ovl_player_actor/z_player.c:5573]: That makes sense. Thanks for the clarification. Should probably add a TODO to verify the MWA though.

Comment (engineer124) [reply src/overlays/actors/ovl_player_actor/z_player.c]: done!

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**